control/main.py

```diff
@@ -0,0 +1,194 @@
+    logger.setLevel(logging.DEBUG)
+    filehandler = logging.FileHandler('main.log')
+    filehandler.setLevel(logging.DEBUG)
+    console = logging.StreamHandler()
+    console.setLevel(logging.DEBUG)
+    formatter = logging.Formatter('%(asctime)s - %(name)s - %(levelname)s - %(message)s')
+    filehandler.setFormatter(formatter)
+    console.setFormatter(formatter)
+    logger.addHandler(filehandler)
+    logger.addHandler(console)
+
+    # Connect to xBee
+    com = Communication(COM_CONNECTION_STRING, 0.1)
+    logger.debug("Connected to wireless communication receiver")
+    com.send(u"Connected to wireless communication receiver")
+
+    # Connect to i2c data server
+    data_client = I2cDataClient("tcp://localhost:5555")
+    if not data_client.read():
+        logger.critical("Can't connect to zmq data server")
+        com.send(u"Can't connect to zmq data server")
+        sys.exit(1)
+
+    # Connect to the drone
+    logger.debug("Starting program, attempting connection to flight controller.")
+    com.send(u"Starting program, attempting connection to flight controller")
+    vehicle_control = None
+    try:
+        vehicle_control = Controller(PIXHAWK_CONNECTION_STRING, baud=57600, com=com)
+        logger.debug("Connected to flight controller")
+        com.send(u"Connected to flight controller")
+    except dronekit.APIException as err:
+        logger.critical("dronekit.APIException: {}".format(err))
+        logger.critical("Could not connect to flight controller.")
+        com.send(u"Could not connect to flight controller.")
+        sys.exit(1)
+
+    # Wait until the waypoints flight path is received from GCS
+    logger.debug("Waiting to receive flight path from GCS")
+    com.send(u"Waiting to receive flight path from GCS")
+    waypoints = com.receive()
+    while not waypoints:
+        waypoints = com.receive()
+
+    # Create points
+    start_location = vehicle_control.vehicle.location.global_relative_frame
+    points = create_waypoints(logger, com, start_location, waypoints)
+
+    if not points:
+        logger.critical("Invalid points received from GCS")
+        com.send(u"Invalid points received from GCS")
+        sys.exit(1)
+
+    # Arm and takeoff
+    vehicle_control.arm()
+    vehicle_control.takeoff(10)
+    points.append(vehicle_control.home)
+
+    # Log points
+    for index, point in enumerate(points):
+        logger.debug("Destination {}: {}".format(index, point))
+
+    # Go to the points
+    if vehicle_control.vehicle.mode.name == "GUIDED":
+        logger.debug("Flying to points...")
+        for point in points:
+            com.send(u"Destination: {}".format(point))
+            if vehicle_control.vehicle.mode.name != "GUIDED":
+                com.send(u"Mode no longer guided")
+                break
+            vehicle_control.goto(point)
+
+            # Wait for vehicle to reach destination before updating the point
+            for sleep_time in range(10):
```

Comment:
You could probably make the range like 60 or something, 60 seconds is probably more than enough time to get to any destination, and 10 seconds is definitely not enough right now